services/TokenUpdate.proto

```diff
@@ -54,5 +53,5 @@ message TokenUpdateTransactionBody {
     Duration autoRenewPeriod = 11; // The new interval at which the auto-renew account will be charged to extend the token's expiry.
     Timestamp expiry = 12; // The new expiry time of the token. Expiry can be updated even if admin key is not set. If the provided expiry is earlier than the current token expiry, transaction wil resolve to INVALID_EXPIRATION_TIME
     google.protobuf.StringValue memo = 13; // If set, the new memo to be associated with the token (UTF-8 encoding max 100 bytes)
-    CustomFees custom_fees = 14; // If set, the new custom fees to be assessed during a CryptoTransfer that transfers units of this token. If the can_update_with_admin_key field is already set to false for the token, this TokenUpdate will resolve to CUSTOM_FEES_ARE_MARKED_IMMUTABLE. If the can_update_with_admin_key field is set to false in this TokenUpdate, the new custom fees will take effect and stay immutable afterwards.
+    Key fee_schedule_key = 14; // If set, the new key to use to update the token's custom fee schedule; if the token does not currently have this key, transaction will resolve to TOKEN_HAS_NO_FEE_SCHEDULE_KEY
```

Comment:
missing `repeated CustomFee custom_fees = 21`

Reply:
it's intended. tokenupdate no longer changes custom fees.

Reply:
Right, `TokenUpdate` can only update the `fee_schedule_key`. The fee schedule itself is updated via the new `TokenFeeScheduleUpdate` transaction. The proto file for this new transaction is missing and that made me confused.

Reply:
Only the new `TokenFeeScheduleUpdate` txn will be able to change the `custom_fees` field.

It's like how only a `TokenMint` or `TokenBurn` can change the token's supply---a `TokenUpdate` can't.